### PR TITLE
[BUG] Fix the scale_factor and pad_param value in LetterResize

### PR DESCRIPTION
**Pull request by tang576225574**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

修复LetterResize在多次resize情况下gt_bboxes结果不对问题

Link issue: #379 

## Modification

只改动了mmyolo/datasets/transforms/transforms.py文件中的LetterResize类：增加一个scale_factor_origin的key记录原来的scale_factor，在运行完transform之后，再利用scale_factor_origin来完成scale_factor的更新

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.
不影响兼容

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by RangeKing** [comment]
Hi @tang576225574, thanks for your contribution. Please click this [Link](https://cla-assistant.io/open-mmlab/mmyolo?pullRequest=387) and sign the CLA.

File: mmyolo/datasets/transforms/transforms.py [hunk 1]
```diff
@@ -210,12 +210,8 @@ def _resize_img(self, results: dict):
         scale_factor = (ratio[1], ratio[0])  # mmcv scale factor is (w, h)
 
         if 'scale_factor' in results:
-            results['scale_factor'] = (results['scale_factor'][0] *
-                                       scale_factor[0],
-                                       results['scale_factor'][1] *
-                                       scale_factor[1])
-        else:
-            results['scale_factor'] = scale_factor
+            results['scale_factor_origin'] = results['scale_factor']
```
Comment (hhaAndroid) [inline mmyolo/datasets/transforms/transforms.py:213]: Thank you very much for your contribution. Can you add a unit test to test this case you are dealing with specifically? I think that would be better.

Comment (tang576225574) [reply mmyolo/datasets/transforms/transforms.py:213]: It is my first time to commit the pull request, that I don't know how to add a unit test....Could you tell me how to do this? Appreciate!

Comment (hhaAndroid) [reply mmyolo/datasets/transforms/transforms.py:213]: https://github.com/open-mmlab/mmyolo/blob/dev/tests/test_datasets/test_transforms/test_transforms.py#L40

You can add new code at the end of this function.

**Comment by hhaAndroid** [comment]
@tang576225574 hi，please fix the lint

**Review by hhaAndroid — APPROVED** [review]

File: mmyolo/datasets/transforms/transforms.py (hunk 1 above)
Comment (tang576225574) [reply mmyolo/datasets/transforms/transforms.py:213]: 抱歉，我刚刚才添加了单元测试，已经提交了，需要重新提pr吗？

**Pull request merged**